indexer/services/ender/src/helpers/notifications/notifications-functions.ts

```diff
@@ -0,0 +1,88 @@
+import { logger, stats } from '@dydxprotocol-indexer/base';
+import {
+  createNotification,
+  NotificationDynamicFieldKey,
+  NotificationType,
+  sendFirebaseMessage,
+} from '@dydxprotocol-indexer/notifications';
+import {
+  OrderFromDatabase,
+  PerpetualMarketFromDatabase,
+  SubaccountFromDatabase,
+  SubaccountTable,
+  FirebaseNotificationTokenTable,
+} from '@dydxprotocol-indexer/postgres';
+
+import config from '../../config';
+
+export async function sendOrderFilledNotification(
+  order: OrderFromDatabase,
+  market: PerpetualMarketFromDatabase,
+) {
+  const start = Date.now();
+  try {
+    const subaccount = await SubaccountTable.findById(order.subaccountId);
+    if (!subaccount) {
+      throw new Error(`Subaccount not found for id ${order.subaccountId}`);
+    }
+
+    const tokens = (await FirebaseNotificationTokenTable.findAll(
+      { address: subaccount.address }, [])
+    );
+    if (tokens.length === 0) {
+      return;
+    }
+
+    const notification = createNotification(
+      NotificationType.ORDER_FILLED,
+      {
+        [NotificationDynamicFieldKey.AMOUNT]: order.size.toString(),
+        [NotificationDynamicFieldKey.MARKET]: market.ticker,
+        [NotificationDynamicFieldKey.AVERAGE_PRICE]: order.price,
+      },
+    );
+
+    await sendFirebaseMessage(tokens, notification);
+  } catch (error) {
+    logger.error({
+      at: 'ender#notification-functions',
+      message: 'Error sending order filled notification',
+      error,
+    });
+  } finally {
+    stats.timing(`${config.SERVICE_NAME}.send_order_filled_notification.timing`, Date.now() - start);
+  }
+}
+
+export async function sendOrderTriggeredNotification(
+  order: OrderFromDatabase,
+  market: PerpetualMarketFromDatabase,
+  subaccount: SubaccountFromDatabase,
+) {
+  const start = Date.now();
+  try {
+    const tokens = (await FirebaseNotificationTokenTable.findAll(
```

Comment:
this is adding an additional db roundtrip. Can you add a TODO to merge this into the conditional order handler sql script?